README.md

```diff
@@ -219,6 +219,32 @@ quarkus.log.category.http-problem.level=ERROR # only HTTP 5XX problems are logge
 quarkus.log.category.http-problem.level=OFF # disables all problems-related logging
 ```
 
+## Custom ProblemPostProcessor
+If you want to intercept, change or augment a mapped `HttpProblem` before it gets serialized into raw HTTP response 
+body, you can create a bean extending `ProblemPostProcessor`, and override `apply` method.
+
+**Important: Make sure your implementation is thread-safe, as it may potentially be called concurrently if multiple 
+requests throw exceptions at the same time.**
+
+Example:
+```java
+@ApplicationScoped
+@Startup // makes sure bean is instantiated eagerly on startup
```

Comment:
Registering CDI beans as postprocessors happens in runtime (RUNTIME_INIT):
```
CDI.current().select(ProblemPostProcessor.class)
                .forEach(ExceptionMapperBase.postProcessorsRegistry::register);
```

If you reference your postprocessor from other beans then this annotation is not needed (Quarkus will create and register it as bean in CDI on startup), but if it's not (and I believe it to be true in most cases) this bean will never be created.
`@Unremovable` - as far as I undestand - does not affect how the bean is instantiated (lazily or eagerly) - from this extension's point of view Quarkus can remove this bean immediately after startup. But the bean needs to be registered in CDI in RUNTIME_INIT phase for this autoregistration to work correctly.

Reply:
> Registering CDI beans as postprocessors happens in runtime (RUNTIME_INIT):
> 
> ```
> CDI.current().select(ProblemPostProcessor.class)
>                 .forEach(ExceptionMapperBase.postProcessorsRegistry::register);
> ```
> 

Sure, my fault.

> If you reference your postprocessor from other beans then this annotation is not needed (Quarkus will create and register it as bean in CDI on startup), but if it's not (and I believe it to be true in most cases) this bean will never be created. 

> `@Unremovable` - as far as I undestand - does not affect how the bean is instantiated (lazily or eagerly) - from this extension's point of view Quarkus can remove this bean immediately after startup. But the bean needs to be registered in CDI in RUNTIME_INIT phase for this autoregistration to work correctly.

Marking a bean ``@Unremovable`` is mentioned in the offical guide especially for that case u have implemented:
Please refer to [Remove unused beans](https://quarkus.io/guides/cdi-reference#remove_unused_beans). 

From my understanding, although the bean looks "unused", it won't get removed. Exactly what we want. Thus a proxy gets instantiated (since it is @ApplicationScoped see [nonstandard features](https://quarkus.io/guides/cdi-reference#nonstandard_features)) and lazily create when registered at ``RUNTIME_INIT``. Otherwise I cannot explain why this code works ..

```
@ApplicationScoped
@Unremovable
public class TestProcessor implements ProblemPostProcessor {
    @Override
    public HttpProblem apply(HttpProblem problem, ProblemContext context) {
        return HttpProblem.builder(problem)
                .with("test", "test")
                .build();
    }
}

```

Reply:
You're right, `@Unremovable` will work as well (I falsely assume removability is about runtime, but in fact it's about build time), I will add this to the example as possible option to use instead of `@Startup`. 

Nevertheless I don't think there's anything particularly wrong with `@Startup`, which makes the bean unremovable (by implicitly declaring an observer method)

Reply:
Yeah, totally agree with that. I've just suggested another idea which could state the intention a little bit more explicitly.

Reply:
Maybe one last idea. How about adding an additional build-step which marks each ``@ProblemPostProcessor`` as unremovable:

```
    @BuildStep
    UnremovableBeanBuildItem unremovableBeans() {
        return UnremovableBeanBuildItem.beanTypes(ProblemPostProcessor.class);
    }
```

By doing so, no additional annotation is required and the user doesn't need to be aware of any Quarkus internals.

Reply:
That looks neat indeed, a tiny bit of magic, but it looks like it's worth it. I'll look into it later.

Reply:
@chberger works like a charm, thanks
https://github.com/TietoEVRY/quarkus-resteasy-problem/commit/c69cc6a36a825a4cdc771dc2ede2620c861659a8